packages/eslint-plugin/docs/rules/no-unsafe-enum-comparison.md

```diff
@@ -6,21 +6,23 @@ description: 'Disallow comparing an enum value with a non-enum value.'
 >
 > See **https://typescript-eslint.io/rules/no-unsafe-enum-comparison** for documentation.
 
-The TypeScript compiler can be surprisingly lenient when working with enums.
-For example, it will allow you to compare enum values against numbers even though they might not have any type overlap:
+The TypeScript compiler can be surprisingly lenient when working with enums. String enums are widely considered to be safer than number enums, but even string enums have some pitfalls. For example, it is allowed to compare enum values against literals:
 
 ```ts
-enum Fruit {
-  Apple,
-  Banana,
+enum Vegetable {
+  Asparagus = 'asparagus',
 }
 
-declare let fruit: Fruit;
+declare const vegetable: Vegetable;
 
-fruit === 999; // No error
+vegetable === 'asparagus'; // No error
 ```
 
-This rule flags when an enum typed value is compared to a non-enum `number`.
+The above code snippet should instead be written as `vegetable === Vegetable.Asparagus`. Allowing literals in comparisons subverts the point of using enums in the first place. By enforcing comparisons with properly typed enums:
+
+- It makes a codebase more resilient to enum members swapping values.
+- It allows for code IDE's to use the "Rename Symbol" feature to quickly rename an enum throughout an entire codebase without anything breaking.
+- It makes enums more intuitive, similar to how they work in other strongly-typed languages like Rust.
```

Comment:
[Content] I don't resonate with this too much. I think it'd be better to talk about the root semantics: that it makes usage of the enums align more to the enum's semantics. Some devs would be resistant to being more akin to strongly typed languages like Rust.

Reply:
do you want to provide an edit suggestion

Reply:
Heh I was hoping you wouldn't ask, I was having a hard time thinking of one... maybe this?

```suggestion
- It aligns code to the proper enum semantics of referring to them by name and treating their values as implementation details.
```